v-model with arguments and defineModel does not work for nullable types since 2.0.28

### Vue - Official extension or vue-tsc version

2.0.28 & 2.0.29

### VSCode version

1.91.1

### Vue version

3.4.35

### TypeScript version

5.5.4

### System Info

```shell
System:
    OS: macOS 14.4.1
    CPU: (8) x64 Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
    Memory: 43.05 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.15.0 - ~/.nvm/versions/node/v20.15.0/bin/node
    npm: 10.8.1 - ~/.nvm/versions/node/v20.15.0/bin/npm
    pnpm: 9.6.0 - ~/.nvm/versions/node/v20.15.0/bin/pnpm
    Watchman: 2024.04.01.00 - /usr/local/bin/watchman
```


### Steps to reproduce

### child.vue
```ts
const msg = defineModel('msg', {
  type: String as PropType<string | null>,
  default: null,
});
```

### parent.vue
```vue
<template>
  <Child v-model:msg="msg" />
</template>
<script lang="ts" setup>
const msg = ref<string | null>('test');
</script>

```

Running `vue-tsc` or in vscode:

```
error TS2322: Type 'string | null' is not assignable to type 'string | undefined'.
163         v-model:msg="msg"
```

For default `v-model` looks like it works, but for [v-model arguments](https://vuejs.org/guide/components/v-model.html#v-model-arguments) it fail.

### What is expected?

It should be correctly inferred as `string | null` and therefore not fail.

### What is actually happening?

In 2.0.28 Error, in 2.0.26 No error

### Link to minimal reproduction

https://stackblitz.com/edit/vitejs-vite-teuagl?file=src%2FApp.vue

No typecheck on stackblitz? :(

### Any additional comments?

Thank you so much for this amazing and invaluable piece of software.

--- a/packages/language-core/lib/codegen/script/scriptSetup.ts
+++ b/packages/language-core/lib/codegen/script/scriptSetup.ts
@@ -413,7 +413,7 @@ function* generateDefinePropType(scriptSetup: NonNullable<Sfc['scriptSetup']>, p
 	}
 	else if ((defineProp.name && defineProp.nameIsString) || !defineProp.nameIsString) {
 		// Infer from actual prop declaration code 
-		yield `NonNullable<typeof ${propName}['value']>`;
+		yield `typeof ${propName}['value']`;
 	}
 	else if (defineProp.defaultValue) {
 		// Infer from defineProp({default: T})

--- a/test-workspace/tsc/vue2/tsconfig.json
+++ b/test-workspace/tsc/vue2/tsconfig.json
@@ -21,6 +21,7 @@
 		"../vue3/#4327",
 		"../vue3/#4512",
 		"../vue3/#4540",
+		"../vue3/#4646",
 		"../vue3/components",
 		"../vue3/defineEmits",
 		"../vue3/defineModel",

--- a/test-workspace/tsc/vue3/#4646/child.vue
+++ b/test-workspace/tsc/vue3/#4646/child.vue
@@ -5,7 +5,7 @@ import {exactType} from 'tsc/shared'
 import {PropType} from 'vue'
 
 const msg = defineModel('msg', {
-    type: String as PropType<string | null>,
+	type: String as PropType<string | null>,
 	required: true,
 })
 

--- a/test-workspace/tsc/vue3/#4646/child.vue
+++ b/test-workspace/tsc/vue3/#4646/child.vue
@@ -1,0 +1,13 @@
+<template>{{ msg }}</template>
+
+<script lang="ts" setup>
+import {exactType} from 'tsc/shared'
+import {PropType} from 'vue'
+
+const msg = defineModel('msg', {
+    type: String as PropType<string | null>,
+	required: true,
+})
+
+exactType(msg.value, {} as string | null)
+</script>

--- a/test-workspace/tsc/vue3/#4646/child2.vue
+++ b/test-workspace/tsc/vue3/#4646/child2.vue
@@ -1,0 +1,5 @@
+<template>{{ other }}</template>
+
+<script lang="ts" setup>
+const other = defineModel<string | null>('other')
+</script>

--- a/test-workspace/tsc/vue3/#4646/parent.vue
+++ b/test-workspace/tsc/vue3/#4646/parent.vue
@@ -1,0 +1,15 @@
+<template>
+  <child v-model:msg="msg" />
+  <child :msg="msg" @update:msg="v => exactType(v, {} as string|null)" />
+
+  <child2 v-model:other="other" />
+</template>
+<script lang="ts" setup>
+import { exactType } from 'tsc/shared'
+import { ref } from 'vue'
+import child from './child.vue'
+import child2 from './child2.vue'
+
+const msg = ref<string | null>('test')
+const other = ref<string | null>('test2')
+</script>

